### PR TITLE
cargo-wasi: init at 0.1.26

### PR DESCRIPTION
**Pull request by lucperkins**
###### Description of changes

This PR adds a package for [`cargo-wasi`](https://github.com/bytecodealliance/cargo-wasi), a `cargo` subcommand for building Rust for the `wasm32-wasi` target.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vojta001** [comment]
Result of `nixpkgs-review pr 186725` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>cargo-wasi</li>
  </ul>
</details>

**Comment by vojta001** [comment]
```
error: builder for '/nix/store/k2p9vcny5bm98rmkya07pikki86d5r0l-cargo-wasi-0.1.26.drv' failed with exit code 101;
last 10 log lines:
>
>
>   It looks like you're compiling on Linux and also targeting Linux. Currently this
>   requires the `pkg-config` utility to find OpenSSL but unfortunately `pkg-config`
>   could not be found. If you have OpenSSL installed you can likely fix this by
>   installing `pkg-config`.
>
>   ', /build/cargo-wasi-0.1.26-vendor.tar.gz/openssl-sys/build/find_normal.rs:191:5
>   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by lucperkins** [comment]
@vojta001 Fixed

**Comment by vojta001** [comment]
Result of `nixpkgs-review pr 186725` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>cargo-wasi</li>
  </ul>
</details>

**Comment by vojta001** [comment]
The error message is still the same. And it seems ofborg came to the same conclusion

**Comment by viraptor** [comment]
Result of `nixpkgs-review pr 186725` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cargo-wasi</li>
  </ul>
</details>

**Comment by lucperkins** [comment]
@vojta001 I'll see if it works in `nativeBuildInputs`

**Comment by vojta001** [comment]
```
  run pkg_config fail: "`\"pkg-config\" \"--libs\" \"--cflags\" \"openssl\"` did not exit successfully: exit status: 1\nerror: could not find system library 'openssl' required by the 'openssl-sys' crate\n\n--- stderr\nPackage openssl was not found in the pkg-config search path.\nPerhaps you should add the directory containing `openssl.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'openssl' found\n"

  --- stderr
  thread 'main' panicked at '

  Could not find directory of OpenSSL installation, and this `-sys` crate cannot
  proceed without this knowledge. If OpenSSL is installed and this crate had
  trouble finding it,  you can set the `OPENSSL_DIR` environment variable for the
  compilation process.

  Make sure you also have the development packages of openssl installed.
  For example, `libssl-dev` on Ubuntu or `openssl-devel` on Fedora.

  If you're in a situation where you think the directory *should* be found
  automatically, please open a bug at https://github.com/sfackler/rust-openssl
  and include information about your system as well as this message.
  ```

**Comment by vojta001** [comment]
I don't have access to full `nixpkgs-review` currently, but running `nix build .#cargo-wasi` succeeds on x86_64-linux

**Review by vojta001 — APPROVED** [review]

**Review by azahi — APPROVED** [review]

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 186725` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cargo-wasi</li>
  </ul>
</details>

File: pkgs/development/tools/rust/cargo-wasi/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    sha256 = "sha256-jugq7A3L+5+YUSyp9WWKBd4BA2pcXKd4CMVg5OVMcEA=";
+  };
+
+  cargoSha256 = "sha256-L4vRLYm1WaCmA4bGyY7D0yxXuqxGSHMMD/wlY8+MgPk=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs =
+    lib.optionals stdenv.isDarwin [ Security ] ++
+    lib.optionals stdenv.isLinux [ openssl ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/cargo-wasi/default.nix:27]:
```suggestion
  buildInputs = lib.optionals stdenv.isLinux [ openssl ]
    ++ lib.optionals stdenv.isDarwin [ Security ];
```

**Pull request merged**